spec/field_types.md

```diff
@@ -11,10 +11,27 @@ _DigestSet (object)_
 > It is RECOMMENDED to use at least `sha256` for compatibility between producers
 > and consumers.
 >
-> Pre-defined algorithm names: `sha256`, `sha224`, `sha384`, `sha512`,
-> `sha512_224`, `sha512_256`, `sha3_224`, `sha3_256`, `sha3_384`, `sha3_512`,
-> `shake128`, `shake256`, `blake2b`, `blake2s`, `ripemd160`, `sm3`, `gost`,
-> `sha1`, `md5`.
+> Pre-defined algorithms:
+>
+> -   `sha256`, `sha224`, `sha384`, `sha512`, `sha512_224`, `sha512_256`,
+>     `sha3_224`, `sha3_256`, `sha3_384`, `sha3_512`, `shake128`, `shake256`,
+>     `blake2b`, `blake2s`, `ripemd160`, `sm3`, `gost`, `sha1`, `md5`
+>
+>     Standard cryptographic hash algorithms, for cases when when the method of
+>     serialization is obvious or well known.
+>
+> -   `goModuleH1`: The go module [directory Hash1], omitting the "h1:"
+>     prefix and output in hexadecimal instead of base64. Can be computed over a
+>     directory named `name@version`, or the contents of zip file containing
+>     such a directory:
+>
+>     ```bash
+>     find name@version -type f | LC_ALL=C sort | xargs -r sha256sum | sha256sum | cut -f1 -d' '
+>     ```
+>
+>     For example, the module dirhash
+>     `h1:47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU=` would be encoded as
+>     `{"goModuleH1": "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"}`.
```

Comment:
To what extent are go developers familiar with the base64 version?

I'm somewhat concerned folks will be surprised that the go reported hash _string_ differs from the hash reported in the provenance.  Would it be acceptable to just state in the docs that this is base64 encoded?  Verifiers will need to use special code to compute this value anyways, so it seems like doing so wouldn't add any complexity.

Reply:
I don't see a good solution here. Line 7 says that the value is always a lowercase hex encoding of the digest. Therefore, our options seem to be:

- Encode it using hex for consistency with all the other algorithms, at the cost of divergence from what go users see. (My suggestion.)
- Change the spec so that encoding is per-algorithm. This is inconsistent with the other fields, and is technically a breaking change but probably OK since no one should be parsing unrecognized fields anyway. (Your suggestion.)
- Make this the hex encoding of the base64, which seems like a bad idea.

Does anyone else have opinions?

Reply:
What if "Pre-defined algorithms" require that it be the lowercase hex encoding, but allow custom values (like this one) to do what makes the most sense in context?  I'm not sure this being a breaking change would be a big problem?  [This note](https://github.com/in-toto/attestation#important) even indicates the contents are subject to change until 1.0.  If needed I think we could bump to 0.2.0.

Perhaps the most important feedback we could get would be from some Go experts.